samples/src/test/java/com/example/bigquery/RelaxTableQueryIT.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.TestCase.assertNotNull;
+
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class RelaxTableQueryIT {
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  private static final String BIGQUERY_PROJECT_ID = System.getenv("BIGQUERY_PROJECT_ID");
+  private static final String BIGQUERY_DATASET_NAME = System.getenv("BIGQUERY_DATASET_NAME");
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("BIGQUERY_PROJECT_ID");
+    requireEnvVar("BIGQUERY_DATASET_NAME");
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+  }
+
+  @Test
+  public void testRelaxTableQuery() throws Exception {
+    String tableName = "RELAX_TABLE_QUERY_TEST";
+    Schema originalSchema =
+        Schema.of(
+            Field.newBuilder("word", LegacySQLTypeName.STRING).setMode(Field.Mode.REQUIRED).build(),
+            Field.newBuilder("word_count", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.REQUIRED)
+                .build(),
+            Field.newBuilder("corpus", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.REQUIRED)
+                .build(),
+            Field.newBuilder("corpus_date", LegacySQLTypeName.STRING)
+                .setMode(Field.Mode.REQUIRED)
+                .build());
+
+    CreateTable.createTable(BIGQUERY_DATASET_NAME, tableName, originalSchema);
+
+    RelaxTableQuery.relaxTableQuery(BIGQUERY_PROJECT_ID, BIGQUERY_DATASET_NAME, tableName);
+    assertThat(bout.toString())
+        .contains("Successfully relaxed all columns in destination table during query job");
+
+    // Clean up
+    DeleteTable.deleteTable(BIGQUERY_DATASET_NAME, tableName);
```

Comment:
nit: should this be in `tearDown`?

Reply:
I don't seem to be able to pass in `tableName` in tearDown() @kurtisvg thoughts?

Reply:
It's probably a good practice because it will ensure the tearDown occurs if the test fails. You should also use a UUID, so that multiple test runs don't step on each other. 

You can accomplish this be having `tableName` be a class field, and using `setUp` to create it and `tearDown` to destroy it.

Reply:
It's probably a good practice because it will ensure the tearDown occurs if the test fails. You should also use a UUID, so that multiple test runs don't step on each other. 

You can accomplish this be having `tableName` be a class field, and using `setUp` to create it and `tearDown` to destroy it.

Reply:
Will do